scripts/publish.js

```diff
@@ -221,6 +222,29 @@ const createStarterDist = async (basename, isTypescript = false) => {
     if (file === "TS-README.md") {
       fs.renameSync(dest, path.join(dir.dist, name, "README.md"))
     }
+    // if we copied over the gatsby-node file, append type-conditional slices config
+    if (file === "gatsby-node.js") {
+      fs.appendFileSync(
+        dest,
+        `
+exports.createPages = ({ actions }) => {
+  const { createSlice } = actions
+  createSlice({
+    id: "header",
+    component: require.resolve("./src/components/header${
+      isTypescript ? ".tsx" : ".js"
```

Comment:
I went this route for adding the config mainly due to the issues of resolving these different file extension variants (js | tsx). If anyone knows a clever way to resolve a tsx or a js file, but this approach also keeps each plugin ignorant of whether it's being used in the context of a TS or JS copy of the repo.